lib/rubocop/cop/rails/file_path.rb

```diff
@@ -120,15 +122,42 @@ def check_for_rails_root_join_with_slash_separated_path(node)
           return unless style == :arguments
           return unless rails_root_nodes?(node)
           return unless rails_root_join_nodes?(node)
-          return unless node.arguments.any? { |arg| string_with_slash?(arg) }
+          return unless valid_slash_separated_path_for_rails_root_join?(node.arguments)
 
           register_offense(node, require_to_s: false) do |corrector|
             autocorrect_rails_root_join_with_slash_separated_path(corrector, node)
           end
         end
 
-        def string_with_slash?(node)
-          node.str_type? && node.source.include?(File::SEPARATOR)
+        def valid_arguments_for_file_join_with_rails_root?(arguments)
+          arguments.any? { |e| rails_root_nodes?(e) } &&
+            arguments.none?(&:variable?) &&
+            arguments.none? { |arg| string_contains_multiple_slashes?(arg) }
+        end
+
+        def valid_string_arguments_for_rails_root_join?(arguments)
+          arguments.size > 1 &&
+            arguments.all?(&:str_type?) &&
+            arguments.none? { |arg| string_with_leading_slash?(arg) } &&
+            arguments.none? { |arg| string_contains_multiple_slashes?(arg) }
+        end
+
+        def valid_slash_separated_path_for_rails_root_join?(arguments)
+          arguments.any? { |arg| string_contains_slash?(arg) } &&
+            arguments.none? { |arg| string_with_leading_slash?(arg) } &&
+            arguments.none? { |arg| string_contains_multiple_slashes?(arg) }
+        end
```

Comment:
Can you refactor to the following?

```suggestion
        def valid_arguments_for_file_join_with_rails_root?(arguments)
          return false unless arguments.any? { |arg| rails_root_nodes?(arg) }

          arguments.none? { |arg| arg.variable? || string_contains_multiple_slashes?(arg) }
        end

        def valid_string_arguments_for_rails_root_join?(arguments)
          return false unless arguments.size > 1
          return false unless arguments.all?(&:str_type?)

          arguments.none? { |arg| string_with_leading_slash?(arg) || string_contains_multiple_slashes?(arg) }
        end

        def valid_slash_separated_path_for_rails_root_join?(arguments)
          return false unless arguments.any? { |arg| string_contains_slash?(arg) }

          arguments.none? { |arg| string_with_leading_slash?(arg) || string_contains_multiple_slashes?(arg) }
        end
```